### ISSUE TITLE
docker-machine support

### ISSUE THREAD

**Issue by sxul** [cross-rs/cross#389]
I'm using docker-machine on Mac OS 10.15.

When I try to run 
`cross build --target x86_64-unknown-linux-gnu` 
I get this error:

```bash
# cross build --target x86_64-unknown-linux-gnu --verbose
+ "rustc" "--print" "sysroot"
+ "rustup" "toolchain" "list"
+ "rustup" "target" "list" "--toolchain" "nightly-x86_64-unknown-linux-gnu"
+ "rustup" "component" "list" "--toolchain" "nightly-x86_64-unknown-linux-gnu"
+ "/usr/local/bin/docker" "run" "--userns" "host" "--rm" "--user" "501:20" "-e" "XARGO_HOME=/xargo" "-e" "CARGO_HOME=/cargo" "-e" "CARGO_TARGET_DIR=/target" "-e" "USER=sxul07" "-e" "CROSS_RUNNER=" "-v" "/Users/sxul07/.xargo:/xargo:Z" "-v" "/Users/sxul07/.cargo:/cargo:Z" "-v" "/cargo/bin" "-v" "/Users/sxul07/Source/hw:/project:Z" "-v" "/Users/sxul07/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu:/rust:Z,ro" "-v" "/Users/sxul07/Source/hw/target:/target:Z" "-w" "/project" "-i" "-t" "rustembedded/cross:x86_64-unknown-linux-gnu-0.2.0" "sh" "-c" "PATH=$PATH:/rust/bin cargo build --target x86_64-unknown-linux-gnu --verbose"
error: failed to open: /target/debug/.cargo-lock

Caused by:
  Permission denied (os error 13)
```

I found a simple solution is replace `"--user" "501:20"` to `"--user" "1000:20"`, and it works fine. 

I don't know how the mac user system works, it looks like I can't find users id 501 and 1000 in /etc/passwd

**Comment by BJK** [comment]
I'm having the same issue. From looking at the code it grabs the UID of the current user. (Pretty sure Mac OS doesn't use /etc/passwd in the traditional way anymore). 

@sxul  Curious how did you change the user ID that is passed? Did you modify the source?

**Comment by sxul** [comment]
> 
> 
> I'm having the same issue. From looking at the code it grabs the UID of the current user. (Pretty sure Mac OS doesn't use /etc/passwd in the traditional way anymore).
> 
> @sxul Curious how did you change the user ID that is passed? Did you modify the source?

yes, modify [here](https://github.com/rust-embedded/cross/blob/5d35b3f33519657ef15446f6d52221a3a9e8e51b/src/docker.rs#L144) and rebuild cross

**Comment by UebelAndre** [comment]
> yes, modify [here](https://github.com/rust-embedded/cross/blob/5d35b3f33519657ef15446f6d52221a3a9e8e51b/src/docker.rs#L144) and rebuild cross

Just a heads up, as of https://github.com/rust-embedded/cross/commit/58bfd65ec083d56065ec2df8a3e7ee0c40fd97a0 you can now use `CROSS_CONTAINER_UID` and `CROSS_CONTAINER_GID` environment variables to solve for the same suggestion.

I've tested this using a github action that runs docker on macos and posted the yaml here: https://github.com/rust-embedded/cross/pull/543#issuecomment-809484573

**Comment by Alexhuszagh** [comment]
I believe this is fixed as of 58bfd65, I will reopen this if it is still an issue. I've added additional documentation on these variables on the [wiki](https://github.com/cross-rs/cross/wiki/Configuration#environment-variables).
